safe_authenticator/src/app_auth.rs

```diff
@@ -218,47 +249,86 @@ fn authenticated_app(
     app: AppInfo,
     app_id: String,
     app_container: bool,
-    _app_permissions: AppPermissions,
+    app_permissions: AppPermissions,
```

Comment:
@Yoga07 shouldn't we be just pulling the issued token for this? We should not need the permission state if the token itself is actually stored.

Reply:
Ah, this is for comparing, as the request _may_ have new permissions in it?

Hmmm. Wonder if we can just pass the token from the prev func here as `potential new token` instead of making another new one here below?